fix(dataset): sort by database in Dataset and Saved queries Issue

rebenitez1802 opened the pull request:
<!---
Please write the PR title following the conventions at https://www.conventionalcommits.org/en/v1.0.0/
Example:
fix(dashboard): load charts correctly
-->

### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Fixes database column sorting in Dataset and SavedQuery list views. After the Ant Design v5 migration, nested columns (like database.database_name) were split into arrays for dataIndex, but when sorting occurs, AntD returns the array format instead of the original dot-notation string, causing backend validation errors.

  The fix converts array fields back to dot notation in the handleTableChange callback while preserving existing string fields unchanged.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TESTING INSTRUCTIONS
<!--- Required! What steps can be taken to manually verify the changes? -->
  1. Navigate to Datasets page (/dataset/list/)
  2. Click on the "Database" column header to sort
  3. Verify sorting works without API errors
  4. Repeat for Saved Queries page (/savedqueryview/list/)
  5. Test sorting other columns to ensure no regressions
  
### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

rebenitez1802 committed:
Added array-to-dot-notation conversion in handleTableChange function

msyavuz commented on superset-frontend/packages/superset-ui-core/src/components/TableCollection/TableCollection.test.tsx:
```diff
@@ -206,3 +206,67 @@ test('Bulk selection should work with pagination', () => {
+    order: 'descend',
+  };
+
+  // Get the component instance to test the callback logic
```
```suggestion
```

msyavuz reviewed (approved):
LGTM with a small nit. Thanks for fixing this!

rebenitez1802 committed:
refactor some test to avoid excesive mocking and avoid using DOM selectors

rebenitez1802 committed:
test fixes

rebenitez1802 committed:
switched to integration test, reimplementation of logic in test removed

msyavuz commented on superset-frontend/packages/superset-ui-core/src/components/TableCollection/TableCollection.test.tsx:
```diff
@@ -206,3 +206,35 @@ test('Bulk selection should work with pagination', () => {
+  expect(firstColumnHeader).toBeInTheDocument();
+
+  // Click on the column header to trigger sorting
+  fireEvent.click(firstColumnHeader);
```
I remember a discussion about usinf `userEvent` vs `fireEvent` but i think this is fine.

msyavuz reviewed (commented):
Test structure is correct now but it tests the wrong column header. The pr fixes sort by database.

rebenitez1802 committed:
added specific values to test sorting with dot-notation accessor to cover database.name use case

rebenitez1802 commented:
@msyavuz I added a generic value to make sure we cover the case scenario that this PR is fixing, I wouldn't use database column because this is a shared component not  reserved for Dataset List, but the value I added follows the same pattern.

rebenitez1802 committed:
small refacto in test to use toHaveBeenCalledWith instead of multiple expects

msyavuz reviewed (approved):
Yes, LGTM!
